draft-ietf-quic-manageability.md

```diff
@@ -905,6 +905,115 @@ need to be flow based.  However, 5-tuple (plus eventually connection ID if
 present) matching is still beneficial for QoS given all packets are handled by
 the same congestion controller.
 
+## QUIC and Network Address Translation (NAT)
+
+QUIC Connection IDs are opaque byte fields that are expressed consistently
+across all QUIC versions {{QUIC-INVARIANTS}}, see {{rebinding}}. This feature
+may appear to present opportunities to optimize NAT port usage and simplify the
+work of the QUIC server. In fact, NAT behavior that relies on CID may instead
+cause connection failure when endpoints change Connection ID, and disable
+important protocol security features. NATs should retain their existing
+4-tuple-based operation and refrain from parsing or otherwise using QUIC
+connection IDs.
+
+This section uses the colloquial term NAT to mean NAPT (section 2.2 of
+{{?RFC3022}}), which overloads several IP addresses to one IP address or to an
+IP address pool, as commonly deployed in carrier-grade NATs or residential NATs.
+
+The remainder of this section explains how QUIC supports NATs better than other
+connection-oriented protocols, why NAT use of Connection ID might appear
+attractive, and how NAT use of CID can create serious problems for the
+endpoints.
+
+{{?RFC4787}} contains some guidance on building NATs to interact constructively
+with a wide range of applications. This section extends the discussion to QUIC.
+
+By using the CID, QUIC connections can survive NAT rebindings as long as no
+routing function in the path is dependent on client IP address and port to
+deliver packets between server and NAT. Reducing the timeout on UDP NATs might
+be tempting in light of this property, but not all QUIC server deployments will
+be robust to rebinding.
+
+### Resource Conservation
+
+NATs sometimes hit an operational limit where they exhaust available public IP
+addresses and ports, and must evict flows from their address/port mapping. CIDs
+might appear to offer a way to multiplex many connections over a single
+address and port.
+
+However, QUIC endpoints may negotiate new connection IDs inside
+cryptographically protected packets, and begin using them at will. Imagine two
+clients behind a NAT that are sharing the same public IP address and port. The
+NAT is differentiating them using the incoming Connection ID. If one client
+secretly changes its connection ID, there will be no mapping for the NAT, and
+the connection will suddenly break.
+
+While mid-connection failure in some cases may seem superior to rejecting QUIC
+outright, HTTP/3 over QUIC falls back to TCP, and
+{{?I-D.ietf-quic-applicability}} recommends TCP fallback for other protocols
+that use QUIC, this is preferable to a connection suddenly black holing and
+timing out, as the recovery time is much longer for a black hole event.
```

Comment:
```suggestion
Some applications might prefer to risk mid-connection failure rather than
not being able to use QUIC. HTTP/3 prefers reliability, recommending that
TCP be used as a fallback and {{?I-D.ietf-quic-applicability}} recommends
TCP fallback for other protocols on the basis that this is preferable to
sudden connection errors and time outs.
```

I had trouble with the phrasing here and I didn't like the comma-splice much.

Reply:
Didn't this completely change the meaning of the first sentence? I don't think it's true that any application would prefer mid-connection failure. The original sentence said "seem superior", meaning that it is not.

Reply:
"seems" is not a good way to phrase a negative; I read it in the positive.  I was trying for a more or less direct translation of the intent.  If you ask me, I think that you could be more assertive about this and say that 

> QUIC is deliberately designed to fail rather than persist when the network cannot support its operation.  For HTTP/3, this extends to recommending a fallback to TCP-based versions of HTTP rather than persisting with a QUIC connection that might be unstable.